service/matching/version_sets.go

```diff
@@ -265,13 +289,18 @@ func extractTargetedVersion(req *workflowservice.UpdateWorkerBuildIdCompatibilit
 		return req.GetPromoteSetByBuildId()
 	} else if req.GetPromoteBuildIdWithinSet() != "" {
 		return req.GetPromoteBuildIdWithinSet()
+	} else if req.GetAddNewBuildIdInNewDefaultSet() != "" {
+		return req.GetAddNewBuildIdInNewDefaultSet()
 	}
-	return req.GetAddNewBuildIdInNewDefaultSet()
+	return req.GetMergeSets().GetPrimarySetBuildId()
 }
 
 // Finds the version in the version sets, returning (set index, index within that set)
 // Returns -1, -1 if not found.
 func findVersion(data *persistencespb.VersioningData, buildID string) (setIndex, indexInSet int) {
+	if buildID == "" {
```

Comment:
When does this happen?

Reply:
If the "secondary" set id isn't specified by the user. It's just a reasonable check in any case.

Reply:
WDYM by "secondary" set ID? this checks `buildID`.

Reply:
That is the path that would be taken to end up calling this with an empty build id.